ProjectBasic/app/src/main/java/com/example/moviedb/screen/navigation/ContainerFragment.kt

```diff
@@ -0,0 +1,137 @@
+
+                        override fun onQueryTextChange(newText: String): Boolean {
+                            //Do Something
+                            return true
+                        }
+
+                        override fun onQueryTextSubmit(query: String): Boolean {
+                            //Do Something
+                            return true
+                        }
+                    })
+                }
+            }
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        if (item.itemId == R.id.ic_searchView_home) {
+            activity?.run {
+                supportFragmentManager.beginTransaction().setCustomAnimations(
+                    R.anim.right_to_left,
+                    R.anim.exit_right_to_left,
+                    R.anim.left_to_right,
+                    R.anim.exit_left_to_right
+                ).replace(R.id.mainFrameLayout, SearchFragment())
+                    .addToBackStack(null)
+                    .commit()
+            }
+        }
+        return false
+    }
+
+    private fun initToolBar() {
+        view?.toolbar?.let {
+            (activity as? MainActivity)?.run { setSupportActionBar(it) }
+        }
+    }
+
+    private fun loadFragment(fragment: Fragment, position: Int): Boolean {
+        activity?.run {
+            when {
+                startingPosition > position -> {
+                    supportFragmentManager.beginTransaction().setCustomAnimations(
+                        R.anim.left_to_right,
+                        R.anim.exit_left_to_right,
+                        R.anim.right_to_left,
+                        R.anim.exit_right_to_left
+                    ).replace(R.id.containerFrameLayout, fragment)
+                        .commit()
+                    changeMenu = true
+                }
+                startingPosition < position -> {
+                    supportFragmentManager.beginTransaction().setCustomAnimations(
+                        R.anim.right_to_left,
+                        R.anim.exit_right_to_left,
+                        R.anim.left_to_right,
+                        R.anim.exit_left_to_right
+                    ).replace(R.id.containerFrameLayout, fragment)
+                        .commit()
+                    changeMenu = false
+                }
+                else -> {
+                    supportFragmentManager.beginTransaction()
+                        .replace(R.id.containerFrameLayout, fragment)
+                        .commit()
+                }
+            }
+            invalidateOptionsMenu()
+        }
+        startingPosition = position
+        return true
+    }
+
+    private val selectedListener =
+        BottomNavigationView.OnNavigationItemSelectedListener { menuItem ->
+            when (menuItem.itemId) {
+                R.id.home -> {
+                    loadFragment(HomeFragment(), 1)
```

Comment:
1: hard code
chú nên tạo 1 enum hoặc annotation class để set cho các thằng item menu này.

```suggestion
                    loadFragment(HomeFragment(), 1)
```

Reply:
sẽ fix ở commit sau